homeassistant/components/gdacs/sensor.py

```diff
@@ -0,0 +1,138 @@
+ATTR_LAST_UPDATE_SUCCESSFUL = "last_update_successful"
+ATTR_LAST_TIMESTAMP = "last_timestamp"
+ATTR_CREATED = "created"
+ATTR_UPDATED = "updated"
+ATTR_REMOVED = "removed"
+
+DEFAULT_UNIT_OF_MEASUREMENT = "alerts"
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the GDACS Feed platform."""
+    manager = hass.data[DOMAIN][FEED][entry.entry_id]
+    sensor = GdacsSensor(entry.entry_id, entry.title, manager)
+    async_add_entities([sensor])
+    _LOGGER.debug("Sensor setup done")
+
+
+class GdacsSensor(Entity):
+    """This is a status sensor for the GDACS integration."""
+
+    def __init__(self, config_entry_id, config_title, manager):
+        """Initialize entity."""
+        self._config_entry_id = config_entry_id
+        self._config_title = config_title
+        self._manager = manager
+        self._status = None
+        self._last_update = None
+        self._last_update_successful = None
+        self._last_timestamp = None
+        self._total = None
+        self._created = None
+        self._updated = None
+        self._removed = None
+        self._remove_signal_status = None
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        self._remove_signal_status = async_dispatcher_connect(
+            self.hass,
+            SIGNAL_STATUS.format(self._config_entry_id),
+            self._update_status_callback,
+        )
+        _LOGGER.debug("Waiting for updates %s", self._config_entry_id)
+        # First update is manual because of how the feed entity manager is updated.
+        await self.async_update()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Call when entity will be removed from hass."""
+        if self._remove_signal_status:
+            self._remove_signal_status()
+
+    @callback
+    def _update_status_callback(self):
+        """Call status update method."""
+        _LOGGER.debug("Received status update for %s", self._config_entry_id)
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def should_poll(self):
+        """No polling needed for GDACS status sensor."""
+        return False
+
+    async def async_update(self):
+        """Update this entity from the data held in the feed manager."""
+        _LOGGER.debug("Updating %s", self._config_entry_id)
+        if self._manager:
+            status_info = self._manager.status_info()
+            if status_info:
+                self._update_from_status_info(status_info)
+
+    def _update_from_status_info(self, status_info):
+        """Update the internal state from the provided information."""
+        self._status = status_info.status
+        self._last_update = (
+            dt.as_utc(status_info.last_update) if status_info.last_update else None
+        )
+        self._last_update_successful = (
+            dt.as_utc(status_info.last_update_successful)
+            if status_info.last_update_successful
```

Comment:
We prefer to use a regular if else block if the ternary operator spans more than one line.

Reply:
OK